Whiskey/Functions/Resolve-WhiskeyTaskPathParameter.ps1

```diff
@@ -0,0 +1,148 @@
+function Resolve-WhiskeyTaskPathParameter
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        # An object that holds context about the current build and executing task.
+        $TaskContext,
+
+        [Parameter(ValueFromPipeline=$true)]
+        [string]
+        $Path,
+
+        [Parameter(Mandatory=$true)]
+        [string]
+        $PropertyName,
+
+        [Parameter(Mandatory=$true)]
+        [Management.Automation.ParameterMetadata]
+        $CmdParameter,
+
+        [Parameter(Mandatory=$true)]
+        [Whiskey.Tasks.ValidatePathAttribute]
+        $ValidateAsPathAttr
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx = -1
+    }
+
+    process
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx++
+
+        $result = $Path
+
+        if( -not $Path -and $ValidateAsPathAttr.Mandatory )
+        {
+            $errorMsg = 'path is mandatory.' -f $TaskProperty[$propertyName]
+            Stop-WhiskeyTask -TaskContext $Context -PropertyName $CmdParameter.Name -Message $errorMsg
+        }
+
+        if( $Path )
+        {
+            if( [IO.Path]::IsPathRooted($Path) -and -not $ValidateAsPathAttr.AllowAbsolute )
+            {
+                Stop-WhiskeyTask -TaskContext $TaskContext -Message ('{0}[{1}] ''{2}'' is absolute but must be relative to the ''{3}'' file.' -f $PropertyName,$pathIdx,$Path,$TaskContext.ConfigurationPath)
+                return
+            }
+
+            if( -not [IO.Path]::IsPathRooted($Path) )
+            {
+                $Path = Join-Path -Path $TaskContext.BuildRoot -ChildPath $Path
+            }
+
+            $optionalParams = @{}
+            if( -not $ValidateAsPathAttr.MustExist)
+            {
+                $optionalParams['ErrorAction'] = 'Ignore'
+            }
+
+            $message = 'Resolve {0} ->' -f $result
+            $prefix = ' ' * ($message.Length - 3)
+            Write-Debug -Message $message
+            $result = 
+                Resolve-Path -Path $Path @optionalParams | 
+                Select-Object -ExpandProperty 'ProviderPath' |
+                ForEach-Object { 
+                    Write-Debug -Message ('{0} -> {1}' -f $prefix,$_)
+                    $_
+                }
+
+            if( -not $result ) 
+            {
+                if( $ValidateAsPathAttr.MustExist -or [WildcardPattern]::ContainsWildcardCharacters( $Path ) )

```

Comment:
The code is currently written so that it will fail if the path did not resolve and: it was supposed to exist, or it contained a wildcard (thus also meaning it needed to exist since `*` implies that the path should resolve to something).